felix/fv/wireguard_test.go

```diff
@@ -1748,6 +1778,7 @@ func wireguardTopologyOptions(routeSource string, ipipEnabled, wireguardIPv4Enab
 	// Indicate wireguard is enabled
 	topologyOptions.WireguardEnabled = wireguardIPv4Enabled
 	topologyOptions.WireguardEnabledV6 = wireguardIPv6Enabled
+	topologyOptions.WireguardThreadingEnabled = wireguardThreadingEnabled
```

Comment:
You need also to remove this, given that the topology option is removed. Please make sure the static check is passing.

Reply:
Thanks - I checked with `make static-checks` within the `felix/fv` directory. This is what I got before removing that line, which was showing no errors:
```
Makefile:602: warning: overriding recipe for target 'golangci-lint'
../lib.Makefile:572: warning: ignoring old recipe for target 'golangci-lint'
make check-fmt golangci-lint
make[1]: Entering directory '/home/debian/calico/felix'
Makefile:602: warning: overriding recipe for target 'golangci-lint'
../lib.Makefile:572: warning: ignoring old recipe for target 'golangci-lint'
Checking code formatting.  Any listed files don't match goimports:
mkdir -p ../.go-pkg-cache bin /home/debian/go/pkg/mod && docker run --rm --net=host --init --init -v /home/debian/calico/felix/../pod2daemon:/go/src/github.com/projectcalico/calico/pod2daemon:rw -v /home/debian/go/pkg/mod:/go/pkg/mod:rw -e LOCAL_USER_ID=1000 -e GOCACHE=/go-cache -e GOARCH=amd64 -e GOPATH=/go -e OS=linux -e GOOS=linux -e GOFLAGS= -v /home/debian/calico:/go/src/github.com/projectcalico/calico:rw -v /home/debian/calico/.go-pkg-cache:/go-cache:rw -w /go/src/github.com/projectcalico/calico/felix calico/go-build:v0.93 bash -c 'exec 5>&1; ! [[ `find . -iname "*.go" ! -wholename "./vendor/*" | xargs goimports -l -local github.com/projectcalico/calico/ | tee >(cat >&5)` ]]'
Starting with UID: 1000
mkdir -p ../.go-pkg-cache bin /home/debian/go/pkg/mod && docker run --rm --net=host --init --init -v /home/debian/calico/felix/../pod2daemon:/go/src/github.com/projectcalico/calico/pod2daemon:rw -v /home/debian/go/pkg/mod:/go/pkg/mod:rw -e LOCAL_USER_ID=1000 -e GOCACHE=/go-cache -e GOARCH=amd64 -e GOPATH=/go -e OS=linux -e GOOS=linux -e GOFLAGS= -v /home/debian/calico:/go/src/github.com/projectcalico/calico:rw -v /home/debian/calico/.go-pkg-cache:/go-cache:rw -w /go/src/github.com/projectcalico/calico/felix  -e CGO_ENABLED=1 -e CGO_LDFLAGS="-L/go/src/github.com/projectcalico/calico/felix/bpf-gpl/include/libbpf/src/amd64 -lbpf -lelf -lz" -e CGO_CFLAGS="-I/go/src/github.com/projectcalico/calico/felix/bpf-gpl/include/libbpf/src -I/go/src/github.com/projectcalico/calico/felix/bpf-gpl" calico/go-build:v0.93 sh -c 'git config --global url."ssh://git@github.com/".insteadOf "https://github.com/"; golangci-lint run --max-issues-per-linter 0 --max-same-issues 0 --timeout 8m'
Starting with UID: 1000
make[1]: Leaving directory '/home/debian/calico/felix'
```

I was able to find the unused variable when I ran `golangci-lint run wireguard_test.go`. It was not picked up with `golangci-lint run ./...`. I'm still debugging my local environment.

Reply:
Thanks. At the Felix level it is returning no errors. I'll debug the E2E tests now.
```
debian@devvm:~/calico/felix$ make static-checks 2>&1
Makefile:602: warning: overriding recipe for target 'golangci-lint'
../lib.Makefile:572: warning: ignoring old recipe for target 'golangci-lint'
make check-fmt golangci-lint
make[1]: Entering directory '/home/debian/calico/felix'
Makefile:602: warning: overriding recipe for target 'golangci-lint'
../lib.Makefile:572: warning: ignoring old recipe for target 'golangci-lint'
Checking code formatting.  Any listed files don't match goimports:
mkdir -p ../.go-pkg-cache bin /home/debian/go/pkg/mod && docker run --rm --net=host --init --init -v /home/debian/calico/felix/../pod2daemon:/go/src/github.com/projectcalico/calico/pod2daemon:rw -v /home/debian/go/pkg/mod:/go/pkg/mod:rw -e LOCAL_USER_ID=1000 -e GOCACHE=/go-cache -e GOARCH=amd64 -e GOPATH=/go -e OS=linux -e GOOS=linux -e GOFLAGS= -v /home/debian/calico:/go/src/github.com/projectcalico/calico:rw -v /home/debian/calico/.go-pkg-cache:/go-cache:rw -w /go/src/github.com/projectcalico/calico/felix calico/go-build:v0.93 bash -c 'exec 5>&1; ! [[ `find . -iname "*.go" ! -wholename "./vendor/*" | xargs goimports -l -local github.com/projectcalico/calico/ | tee >(cat >&5)` ]]'
Starting with UID: 1000
mkdir -p ../.go-pkg-cache bin /home/debian/go/pkg/mod && docker run --rm --net=host --init --init -v /home/debian/calico/felix/../pod2daemon:/go/src/github.com/projectcalico/calico/pod2daemon:rw -v /home/debian/go/pkg/mod:/go/pkg/mod:rw -e LOCAL_USER_ID=1000 -e GOCACHE=/go-cache -e GOARCH=amd64 -e GOPATH=/go -e OS=linux -e GOOS=linux -e GOFLAGS= -v /home/debian/calico:/go/src/github.com/projectcalico/calico:rw -v /home/debian/calico/.go-pkg-cache:/go-cache:rw -w /go/src/github.com/projectcalico/calico/felix  -e CGO_ENABLED=1 -e CGO_LDFLAGS="-L/go/src/github.com/projectcalico/calico/felix/bpf-gpl/include/libbpf/src/amd64 -lbpf -lelf -lz" -e CGO_CFLAGS="-I/go/src/github.com/projectcalico/calico/felix/bpf-gpl/include/libbpf/src -I/go/src/github.com/projectcalico/calico/felix/bpf-gpl" calico/go-build:v0.93 sh -c 'git config --global url."ssh://git@github.com/".insteadOf "https://github.com/"; golangci-lint run --max-issues-per-linter 0 --max-same-issues 0 --timeout 8m'
Starting with UID: 1000
make[1]: Leaving directory '/home/debian/calico/felix'
```